File: app/src/main/java/edu/uoc/gruizto/mybooks/db/BookDao.java
package edu.uoc.gruizto.mybooks.db;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface BookDao {
    @Query("SELECT * FROM mybooks ORDER BY id ASC")
    List<Book> getAll();

    @Query("SELECT * FROM mybooks WHERE id=:id LIMIT 1")
    Book findById(String id);

    @Insert
    void insert(Book book);

    @Delete
    void delete(Book book);

    @Query("DELETE FROM mybooks")
    void deleteAll();
}

File: app/src/main/java/edu/uoc/gruizto/mybooks/db/BookDao.kt
package edu.uoc.gruizto.mybooks.db

import androidx.lifecycle.LiveData
import androidx.room.*
import io.reactivex.Completable
import io.reactivex.Maybe

@Dao
interface BookDao {
    @get:Query("SELECT * FROM mybooks ORDER BY id ASC")
    val all: LiveData<List<Book>>

    @Query("SELECT * FROM mybooks WHERE id=:id LIMIT 1")
    fun findById(id: String): Maybe<Book>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insert(book: Book): Completable

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insertMany(books: List<Book>): Completable
    
    @Delete
    fun delete(book: Book): Completable

    @Query("DELETE FROM mybooks")
    fun deleteAll(): Completable
}
